### PR TITLE
Add ability to set subject name strategies

### PR DESCRIPTION
**Pull request by mpyatishev**
Unfortunately, PR #502 stalled for a long time. So I decided to make my humble attempt to solve this problem.
With this patch, you can set the Subject Name Strategy in CachedSchemaRegistryClient , for example:
```
client = CachedSchemaRegistryClient({
  'url': 'http://localhost',
  'key.subject.name.strategy': "TopicRecordNameStrategy",
  'value.subject.name.strategy': "TopicRecordNameStrategy",
})
```

### CODE REVIEW

**Comment by mpyatishev** [comment]
[clabot:check]

**Comment by mpyatishev** [comment]
[clabot:check]

**Comment by mpyatishev** [comment]
Can anybody review this PR?

File: tests/avro/test_message_serializer.py [hunk 1]
```diff
@@ -80,5 +81,23 @@ def test_decode_none(self):
...
+    def test__get_subject_for_key(self):
+        basic = avro.loads(data_gen.BASIC_SCHEMA)
+        topic = "topic"
+        self.ms.registry_client.key_subject_name_strategy_func = topic_record_name_strategy  # noqa
+        subject = self.ms._get_subject(topic=topic, schema=basic, is_key=True)
+
+        expected = "topic-basic-key"
+        self.assertEqual(expected, subject)
+
+    def test__get_subject_for_value(self):
```
Comment (rnpridgeon) [inline tests/avro/test_message_serializer.py]: Add tests for the other strategies as well.

Comment (mpyatishev) [reply tests/avro/test_message_serializer.py]: I will do it

File: confluent_kafka/avro/serializer/message_serializer.py [hunk 2]
```diff
@@ -85,6 +85,16 @@ def _get_encoder_func(self, writer_schema):
         writer = avro.io.DatumWriter(writer_schema)
         return lambda record, fp: writer.write(record, avro.io.BinaryEncoder(fp))
 
+    def _get_subject(self, topic, schema, is_key):
+        if is_key:
+            subject_suffix = '-key'
+            # get the latest schema for the subject
+            subject = self.registry_client.key_subject_name_strategy_func(topic, schema) + subject_suffix  # noqa
```
[the remark this answers is not in the document]
Comment (mpyatishev) [reply confluent_kafka/avro/serializer/message_serializer.py]: Hm, I see. Hence, schemas for key and value in *RecordNameStrategy will have the same subject name?

File: tests/avro/test_message_serializer.py (hunk 1 above)
Comment (mpyatishev) [reply tests/avro/test_message_serializer.py]: Is tests in https://github.com/confluentinc/confluent-kafka-python/pull/764/files#diff-057a19a5cfd9b6c25d4809ce6e12cf62 not enough?

Comment (rnpridgeon) [reply tests/avro/test_message_serializer.py]: Not when you mutate the subject name outside of the subject name strategy function as you do now(appending the message field suffix in the message serializer). I'm more concerned with the final subject name than the intermediate representation. In the end my schema will be registered under this namespace so this is where we want to test for regressions.

**Comment by mpyatishev** [comment]
How can I restart the AppVeyor build if the problem is not in my code?

**Comment by mpyatishev** [comment]
@rnpridgeon , can you review my fixes for your changes request?

**Comment by agates4** [comment]
@mpyatishev thank you very much for this PR. I hope that this gets approved and merged. This is a problem that is blocking our org.

**Comment by agates4** [comment]
@rnpridgeon please review

**Comment by agates4** [comment]
@edenhill please review

**Comment by edenhill** [comment]
I believe @rnpridgeon 's PR #787 will add support for this in some form.

**Comment by mpyatishev** [comment]
It seems like I should close this PR in favor of #787

**Pull request closed without merging**